### PR TITLE
Hydra e2e test

### LINKED ISSUE #79: Fix `move to hydra` StandingBid redeemer and make it check Hydra head
https://github.com/mlabs-haskell/hydra-auction/blob/c8f53dfc32625f8b4e49d3388d64444c4f29b67b/doc/spec.md#standing-bid-validator

Blocked by #125

### LINKED ISSUE #124: Add stub for hydra version of Ledger tests (with Hydra node started)
Could be left empty, apart for Hydra starting, I think.

### LINKED ISSUE #144: Make bids on L2
Blocked by #125 

CLI is in separate task.

### PR DESCRIPTION
**Pull request by uhbif19**
Closes #124 
Closes #144 
Closes #79 

Blocked by #240 

CI broken by #249

### CODE REVIEW

File: src/HydraAuction/Tx/StandingBid.hs [hunk 1]
```diff
@@ -85,23 +100,38 @@ getStadingBidDatum standingBidUtxo =
...
+  MkExecutionContext {actor} <- ask
+  newBid' terms actor bidAmount
+
+newBid' ::
+  (MonadIO m, MonadFail m, MonadCardanoClient m, MonadTrace m) => AuctionTerms -> Actor -> Natural -> m ()
+newBid' terms submitingActor bidAmount = do
+  -- Actor is not neccesary bidder, on L2 it may be commiter
+  logMsg "Doing new bid"
+
+  (bidderAddress, bidderVk, bidderSk) <- addressAndKeysForActor submitingActor
```
Comment (nazrhom) [inline src/HydraAuction/Tx/StandingBid.hs]: Maybe we can call these `submitterAddress, submitterVk,...`?

Comment (uhbif19) [reply src/HydraAuction/Tx/StandingBid.hs]: You right, I forgot about that.

File: src/HydraAuction/OnChain/Common.hs [hunk 2]
```diff
@@ -53,13 +55,13 @@ stageToInterval terms stage = case stage of
...
 checkInterval terms stage info =
-  traceIfFalse "Wrong interval for transaction" $
+  traceIfFalse "Wrong interval for transaction (checkInterval)" $
     contains (stageToInterval terms stage) (txInfoValidRange info)
 
 {-# INLINEABLE checkVoucherExpiredOrLater #-}
 checkVoucherExpiredOrLater :: AuctionTerms -> TxInfo -> Bool
 checkVoucherExpiredOrLater terms info =
-  traceIfFalse "Wrong interval for transaction" $
+  traceIfFalse "Wrong interval for transaction (checkcheckVoucherExpiredOrLaterVoucherExpiredOrLater)" $
```
Comment (GeorgeFlerovsky) [inline src/HydraAuction/OnChain/Common.hs]: Typo?
```suggestion
  traceIfFalse "Wrong interval for transaction (checkcheckVoucherExpiredOrLater)" $
```

File: docker-compose.yaml [hunk 3]
```diff
@@ -38,6 +38,7 @@ services:
       , "--peer", "hydra-node-2:5001"
       , "--peer", "hydra-node-3:5001"
       , "--hydra-scripts-tx-id", "${HYDRA_SCRIPTS_TX_ID}"
+      , "--contestation-period", "1"
```
Comment (GeorgeFlerovsky) [inline docker-compose.yaml:41]: Only 1 second for contestation?

File: src/HydraAuction/Tx/StandingBid.hs [hunk 4]
```diff
@@ -85,28 +100,43 @@ getStadingBidDatum standingBidUtxo =
...
     Nothing -> Nothing
 
-getApprovedBidders :: AuctionTerms -> Runner ApprovedBidders
+getApprovedBidders :: (MonadNetworkId m, MonadQueryUtxo m) => AuctionTerms -> m ApprovedBidders
 getApprovedBidders terms = do
   standingBidUtxo <- scriptUtxos StandingBid terms
   let StandingBidDatum {standingBidState} = getStadingBidDatum standingBidUtxo
   pure $ approvedBidders standingBidState
 
 newBid :: AuctionTerms -> Natural -> Runner ()
```
Comment (GeorgeFlerovsky) [inline src/HydraAuction/Tx/StandingBid.hs:117]: Why are some type signatures using the concrete `Runner` type, while others use the `(MonadNetworkId m, MonadQueryUtxo m)` class constraints?

Would it be better to use the class constraints uniformly throughout?

**Review by GeorgeFlerovsky — COMMENTED** [review]
Generally looks good, but I cannot approve the PR because I can't run the test suite.
(docker-compose doesn't mount the volumes when I run it on my laptops)

**Comment by uhbif19** [comment]
@GeorgeFlerovsky 

> Generally looks good, but I cannot approve the PR because I can't run the test suite.

This could be delegated to one who clicks `Merge` button.

I re-checked tests locally, so gonna merge.

File: docker-compose.yaml (hunk 3 above)
Comment (uhbif19) [reply docker-compose.yaml:41]: This is used for E2E tests, so that way it works fast.

File: src/HydraAuction/Tx/StandingBid.hs (hunk 4 above)
Comment (uhbif19) [reply src/HydraAuction/Tx/StandingBid.hs:117]: Because `Runner` has `actor`, which is not abstracted yet. This is planned in some tech-debt isseu.

**Pull request merged**